pkg/archive/paths.go

```diff
@@ -17,18 +17,48 @@ package archive
 import (
 	"archive/tar"
 	"archive/zip"
+	"compress/bzip2"
 	"compress/gzip"
 	"context"
 	"io"
 	"os"
+	"strings"
 )
 
-// A WalkFn iterates through an archive, calling FileWalkFn on each member file.
-type WalkFn func(ctx context.Context, path string, walkFn FileWalkFn) error
+const (
+	TarArchive FormatType = iota
+	TarGzArchive
+	TarBz2Archive
+	ZipArchive
+	UnsupportedArchive
+)
+
+var (
+	extensions = map[string]FormatType{
+		"ear":     ZipArchive,
+		"jar":     ZipArchive,
+		"par":     ZipArchive,
+		"war":     ZipArchive,
+		"zip":     ZipArchive,
+		"tar":     TarArchive,
+		"tar.gz":  TarGzArchive,
+		"tgz":     TarGzArchive,
+		"tar.bz2": TarBz2Archive,
+		"tbz2":    TarBz2Archive,
+	}
+)
```

Comment:
Shall we move these into own file `formats.go` file along with the `CheckArchiveType` function?